### PR TITLE
Make fuel tanks explodey again

### PR DESCRIPTION
**Pull request by ElectroJr**
## About the PR

This makes fuel tanks explode when taking heat damage. Shooting them with a laser gun will cause them to explode, a normal gun will just destroy & spill contents.

## The issue
Previously, anything with an `ExplosiveComponent` (like fuel tanks) were supposed to explode when being destroyed. However, the `TriggerSystem` was listening for a `DestructionEventArgs` event, which is never actually raised by `ActSystem` (currently it only looks for components that implement `IDestroyAct`).

## Changes
Instead of just making `ActSystem` raise the event, I added a damage threshold `ExplodeBehavior`. This way you can have explosives that are not destruction sensitive. E.g., some sort of stable explosive that needs to be triggered intentionally by a triggering setup and wont just explode when punched by an assistant. Other than fuel tanks, all explodable entities are already triggered by timers or on-collision, so only fuel tanks have this new `ExplodeBehavior`.

If you want your explosive to always explode when destroyed, just add the `ExplodeBehavior` wherever the destruction act behaviour is defined.

## Other changes.
Previously only the full fuel tank prototype was explosive. AFAIK this means that if you spawned in an empty tank, and manually filled it up it would stay non-explosive. I've just made all fuel tanks explosive (idk, fuel vapours or something?). 

I dream of a day where we can remove `ExplosiveComponent` altogether and just have explosion behaviour based on solutions & chemical reactions.

:cl:
- fix: Fixed fuel tanks not exploding when damaged. They now explode when taking heat damage.

### CODE REVIEW

File: Resources/Prototypes/Entities/Structures/Storage/Tanks/tanks.yml [hunk 1]
```diff
@@ -12,6 +12,14 @@
     state: fueltank
   - type: ReagentTank
     tankType: Fuel
+  - type: Explosive
+    # The explosive component is independent of the stored liquid :/
+    # I dream of a day where all grenades/explosives will be reagent dependent.
+    # For now let's pretend it is left over fuel vapours or something, so empty tanks can still explode.
```
Comment (PaulRitter) [inline Resources/Prototypes/Entities/Structures/Storage/Tanks/tanks.yml:18]: fun fact: empty tanks explode. full tanks usually just burn. (obv depends on the liquid inside, im talking the average gas for your car here)

File: Content.Server/Explosion/Components/ExplosiveComponent.cs [hunk 2]
```diff
@@ -1,11 +1,21 @@
...
-    /// Specifies an explosion range should this entity be exploded.
+    ///     Specifies an explosion range should this entity be exploded.
     /// </summary>
+    /// <remarks>
+    ///     Explosions can be caused by:
+    ///     <list type="bullet">
+    ///         <item>Reaching a damage threshold that causes a <see cref="ExplodeBehavior"/></item>
+    ///         <item>Being triggered via the <see cref="ExplodeOnTriggerComponent"/></item>
+    ///         <item>Manually by some other system via functions in <see cref="ExplosionHelper"/> (for example, see
+    ///         <see cref="WelderComponent"/>'s interaction with fuel tanks).</item>
```
Comment (PaulRitter) [inline Content.Server/Explosion/Components/ExplosiveComponent.cs:16]: i'd like for weldercomponent to not directly invoke this, but for it to just apply heat damage and have that triggering the explosion.

**Review by PaulRitter — CHANGES_REQUESTED** [review]

File: Resources/Prototypes/Entities/Structures/Storage/Tanks/tanks.yml (hunk 1 above)
Comment (ElectroJr) [reply Resources/Prototypes/Entities/Structures/Storage/Tanks/tanks.yml:18]: I realise that this sort of behaviour would complicate any eventual plans to replace explosive component with chemical-reagent based explosions, given that this becomes explosive as you have LESS reagent.

Other than just making fuel tanks a special case, I guess the way to generalise that sort of behaviour would be to either:
- Add an "oxygen" reagent to the tank as it is emptied to simulate air+fuel-vapour interactions, and allow solutions with some minimum fuel:oxygen ratio to explode when ignited
- Require atmos-chemistry with explosive fuel-vapour + air mixtures.

File: Content.Server/Explosion/Components/ExplosiveComponent.cs (hunk 2 above)
Comment (ElectroJr) [reply Content.Server/Explosion/Components/ExplosiveComponent.cs]: Yeah that actually makes more sense.

I've removed the direct call to SpawnExplosion and instead just added a DamageOnToolInteract component to fuel tanks, so they take damage when interacted with using a lit welder.

File: Resources/Prototypes/Entities/Structures/Storage/Tanks/tanks.yml (hunk 1 above)
Comment (PaulRitter) [reply Resources/Prototypes/Entities/Structures/Storage/Tanks/tanks.yml]: first sounds fine, altough as it is rn its actually fine tbh

**Review by Zumorica — APPROVED** [review]

**Pull request merged**